buildSrc/src/main/groovy/airbyte-standard-source-test-file.gradle

```diff
@@ -26,32 +28,37 @@ class AirbyteStandardSourceTestFilePlugin implements Plugin<Project> {
 
         project.task('standardSourceTestFile') {
             doFirst {
-                project.exec {
-                    validateConfig(config)
-                    def targetMountDirectory = "/test_input"
-                    def args = [
-                            'docker', 'run', '--rm', '-i',
-                            // provide access to the docker daemon
-                            '-v', "/var/run/docker.sock:/var/run/docker.sock",
-                            // A container within a container mounts from the host filesystem, not the parent container.
-                            // this forces /tmp to be the same directory for host, parent container, and child container.
-                            '-v', "/tmp:/tmp",
-                            // mount the project dir. all provided input paths must be relative to that dir.
-                            '-v', "${project.projectDir.absolutePath}:${targetMountDirectory}",
-                            '--name', "std-fs-source-test-${project.name}",
-                            'airbyte/base-standard-source-test-file:dev',
-                            '--imageName', DockerHelpers.getDevTaggedImage(project.projectDir, 'Dockerfile'),
-                            '--catalog', "${targetMountDirectory}/${config.configuredCatalogPath}",
-                            '--spec', "${targetMountDirectory}/${config.specPath}",
-                            '--config', "${targetMountDirectory}/${config.configPath}",
-                    ]
+                config.prehook.call()
```

Comment:
Why not put this in the `try`? Seems like the prehook may do some things that need to be cleaned up even if it fails.

Reply:
I am getting super nervous by the amount of logic we put in gradle at the moment.

What is the longer term vision here?

Reply:
why isn't his running in the docker container that you're running instead of in gradle / outside of a controlled docker environment?

Reply:
The docker container run here executes tests which are configured via files. In case we need to say, run and stop a docker container in pre/post hooks (in this case used for Mongo testing), this seems like the most ergonomic way to declare pre/post hooks which start/stop the DB image. These hooks are better used to run docker images, not arbitrary code. I could change the interface to say, take as input a docker image instead of run freeform code, but I think this gets wonky if the user wants to pass options to the `docker` command. 

We could declare these in language-specific bindings (similar to what we do in python) but this is really meant as an escape hatch for languages for which we don't have those bindings (in this case Ruby).